model-optimizer/extensions/ops/gathernd_test.py

```diff
@@ -0,0 +1,257 @@
+        res_output_value = graph.node['output']['value']
+
+        self.assertTrue(np.array_equal(output4, res_output_value),
+                        'values do not match expected: {} and given: {}'.format(output4, res_output_value))
+
+    def test_infer5(self):
+        graph = build_graph(nodes_attributes, edges, inputs5)
+        gathernd_node = Node(graph, 'gathernd_node')
+        GatherND.infer(gathernd_node)
+
+        # get the result
+        res_output_value = graph.node['output']['value']
+
+        self.assertTrue(np.array_equal(output5, res_output_value),
+                        'values do not match expected: {} and given: {}'.format(output4, res_output_value))
+
+    def test_infer6(self):
+        nodes_attributes['gathernd_node']['batch_dims'] = 1
+        graph = build_graph(nodes_attributes, edges, inputs6)
+        gathernd_node = Node(graph, 'gathernd_node')
+        GatherND.infer(gathernd_node)
+
+        # get the result
+        res_output_value = graph.node['output']['value']
+
+        self.assertTrue(np.array_equal(output6, res_output_value),
+                        'values do not match expected: {} and given: {}'.format(output4, res_output_value))
+        nodes_attributes['gathernd_node']['batch_dims'] = 0
+
+    def test_infer7(self):
+        nodes_attributes['gathernd_node']['batch_dims'] = 2
+        graph = build_graph(nodes_attributes, edges, inputs7)
+        gathernd_node = Node(graph, 'gathernd_node')
+        GatherND.infer(gathernd_node)
+
+        # get the result
+        res_output_value = graph.node['output']['value']
+
+        self.assertTrue(np.array_equal(output7, res_output_value),
+                        'values do not match expected: {} and given: {}'.format(output4, res_output_value))
+        nodes_attributes['gathernd_node']['batch_dims'] = 0
+
+    def test_infer8(self):
+        nodes_attributes['gathernd_node']['batch_dims'] = 2
+        graph = build_graph(nodes_attributes, edges, inputs8)
+        gathernd_node = Node(graph, 'gathernd_node')
+        GatherND.infer(gathernd_node)
+
+        # get the result
+        res_output_value = graph.node['output']['value']
+
+        self.assertTrue(np.array_equal(output8, res_output_value),
+                        'values do not match expected: {} and given: {}'.format(output4, res_output_value))
+        nodes_attributes['gathernd_node']['batch_dims'] = 0
+
+    def test_infer_invalid1(self):
+        graph = build_graph(nodes_attributes, edges, inputs_inv1)
+        gathernd_node = Node(graph, 'gathernd_node')
+        self.assertRaises(AssertionError, GatherND.infer, gathernd_node)
+
+    def test_infer_invalid2(self):
+        nodes_attributes['gathernd_node']['batch_dims'] = 2
+        graph = build_graph(nodes_attributes, edges, inputs_inv2)
+        gathernd_node = Node(graph, 'gathernd_node')
+        self.assertRaises(AssertionError, GatherND.infer, gathernd_node)
+        nodes_attributes['gathernd_node']['batch_dims'] = 0
```

Comment:
May be move this line to `SetUp` method? Or create a copy of `nodes_attributes` in the start of `setUp` method and update it instead of modifying  the global `nodes_attributes` variable.

The same comment for other tests.

Reply:
Fixed. Thanks for this way.